consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/PayloadIdentifier.java

```diff
@@ -56,18 +59,27 @@ public PayloadIdentifier(final Long payloadId) {
    * @param timestamp the timestamp
    * @param prevRandao the prev randao
    * @param feeRecipient the fee recipient
+   * @param withdrawals the withdrawals
    * @return the payload identifier
    */
   public static PayloadIdentifier forPayloadParams(
       final Hash parentHash,
       final Long timestamp,
       final Bytes32 prevRandao,
-      final Address feeRecipient) {
+      final Address feeRecipient,
+      final Optional<List<Withdrawal>> withdrawals) {
+
     return new PayloadIdentifier(
         timestamp
             ^ ((long) parentHash.toHexString().hashCode()) << 8
             ^ ((long) prevRandao.toHexString().hashCode()) << 16
-            ^ ((long) feeRecipient.toHexString().hashCode()) << 24);
+            ^ ((long) feeRecipient.toHexString().hashCode()) << 24
+            ^ (long)
+                withdrawals
+                    .map(
+                        ws ->
+                            ws.stream().map(Withdrawal::hashCode).reduce(0, (a, b) -> a ^ (b * 31)))
```

Comment:
great place for a clarifying comment

Reply:
Which bit should I clarify?

I think the forPayloadParams method maybe needs a comment about strategy for creating the hash, but I'm missing the original context.

I'm also concerned about the even numbers used in the bitshifting of the other hashes, e.g. `(long) parentHash.toHexString().hashCode()) << 8`
I found collisions when I did something similar with withdrawals.

Reply:
It might be wise to sort before mapping the list, on the off chance that the order is different but the contents the same.

Reply:
Withdrawals should be in the same order, but good idea to sort the list just in case as we can't guarantee it at this level